internal/pkg/cli/task_run_test.go

```diff
@@ -563,16 +583,25 @@ func TestTaskRunOpts_Execute(t *testing.T) {
 
 			opts := &runTaskOpts{
 				runTaskVars: runTaskVars{
+					GlobalOpts: &GlobalOpts{},
 					groupName: inGroupName,
 					image:     tc.inImage,
 					imageTag:  tc.inTag,
+					env:       tc.inEnv,
 				},
 				spinner:  &mockSpinner{},
-				deployer: mockDeployer,
 			}
 			opts.configureRuntimeOpts = func() error {
-				opts.repository = mockRepo
 				opts.runner = mockRunner
+				opts.deployer = mockDeployer
+				opts.configureRepository = func() error {
+					opts.repository = mockRepo
+					return nil
+				}
+
+				if opts.env != "" {
+					opts.deployOpts = []awscloudformation.StackOption{awscloudformation.WithRoleARN("execution role")}
+				}
```

Comment:
Mmm we shouldn't have custom logic in our tests otherwise we are not testing the real code-path. 
I wonder if instead we should store the `targetEnv` as a field and have that lookup in the `deploy()` method so that we test if the real code checks if an env is not nil and passes the appropriate execution role arn.

I really like the two test-cases written above :)

Reply:
If `targetEnv` is being set in `configureRuntimeOpts`, it seems like we will still not able to set it without writing custom logic in the tests, right?

It seems to me that in order to make the testing work, I will have to somehow isolate the `targetEnv`-related logic from `configureRuntimeOpts`. 

This is reflected in the latest code change. I'm not sure if I like this approach - please let me know what you think👀

Reply:
jes totally makes sense, thank you!